drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -298,7 +284,28 @@ static int32_t optimize_regulator_voltage_scale(uint32_t sysclk_freq)
 	defined(CONFIG_CLOCK_STM32_PLL_SRC_HSI) || \
 	defined(CONFIG_CLOCK_STM32_PLL_SRC_CSI)
 
-static int32_t get_vco_output_range(uint32_t vco_input_range)
+static int get_vco_input_range(uint32_t m_div, uint32_t *range)
+{
+	uint32_t vco_freq;
+
+	vco_freq = PLLSRC_FREQ / m_div;
+
+	if (1000000UL <= vco_freq && vco_freq <= 2000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_1_2;
+	} else if (2000000UL < vco_freq && vco_freq <= 4000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_2_4;
+	} else if (4000000UL < vco_freq && vco_freq <= 8000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_4_8;
+	} else if (8000000UL < vco_freq && vco_freq <= 16000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_8_16;
+	} else {
+		return -ERANGE;
+	}
+
+	return 0;
```

Comment:
Can you move this part to a dedicated commit/PR ?

Reply:
done